tests/models/test_requests.py

```diff
@@ -10,7 +10,7 @@ def test_request_repr():
 
 def test_no_content():
     request = httpx.Request("GET", "http://example.org")
-    assert "Content-Length" not in request.headers
+    assert request.headers["Content-Length"] == "0"
```

Comment:
We shouldn't send Content-Length with get requests
References:
 - https://github.com/encode/httpx/pull/866#issuecomment-605899948
 - https://tools.ietf.org/html/rfc7230#section-3.3.2 "A user agent SHOULD NOT ..."
 - `requests` implementation for non-stream requests https://github.com/psf/requests/blob/fd13816d015c4c90ee65297fa996caea6a094ed1/requests/models.py#L522-L533
 - `requests` implementation for stream requests https://github.com/psf/requests/blob/fd13816d015c4c90ee65297fa996caea6a094ed1/requests/models.py#L476-L479